src/OpenFeature.Extensions.Hosting/Internal/OpenFeatureHostedService.cs

```diff
@@ -0,0 +1,36 @@
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+
+namespace OpenFeature.Internal;
+
+/// <summary>
+///
+/// </summary>
+public sealed class OpenFeatureHostedService(Api api, IEnumerable<FeatureProvider> providers) : IHostedLifecycleService
+{
+    readonly Api _api = Check.NotNull(api);
+    readonly IEnumerable<FeatureProvider> _providers = Check.NotNull(providers);
+
+    async Task IHostedLifecycleService.StartingAsync(CancellationToken cancellationToken)
+    {
+        foreach (var provider in this._providers)
+        {
+            await this._api.SetProviderAsync(provider.GetMetadata().Name ?? string.Empty, provider).ConfigureAwait(false);
```

Comment:
I don't think the meta-data name for a provider is an appropriate name here. We probably need a way to make providers  that encapsulates that data.

Simple example is using one provider with two different environments.

But more than that the provider name is a different domain of names. It is the domain of the provider implementation, where client names are in the domain of the application. An application developer needs to know and assign that name.

Reply:
:100: